src/dsp_tools/commands/xmlupload/models/value_transformers.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import json
+from abc import ABC
+from abc import abstractmethod
+from typing import TypeAlias
+from typing import Union
+from typing import assert_never
+
+from dsp_tools.commands.xmlupload.models.formatted_text_value import FormattedTextValue
+from dsp_tools.models.exceptions import BaseError
+from dsp_tools.utils.date_util import Date
+from dsp_tools.utils.date_util import parse_date_string
+
+InputTypes: TypeAlias = Union[str, FormattedTextValue]
+OutputTypes: TypeAlias = Union[str, Date]
+
+
+class ValueTransformer(ABC):
+    """Class used to transform an input value."""
+
+    @abstractmethod
+    def transform(self, input_value: InputTypes) -> OutputTypes: ...
+
+    @staticmethod
+    def _assert_is_string(value: str | FormattedTextValue) -> str:
+        match value:
+            case str() as s:
+                return s
+            case FormattedTextValue() as xml:
+                raise BaseError(f"Expected string value, but got XML value: {xml.as_xml()}")
+            case _:
+                assert_never(value)
+
+
+class DateTransformer(ValueTransformer):
+    def transform(self, input_value: InputTypes) -> Date:
+        val = self._assert_is_string(input_value)
+        return parse_date_string(val)
+
+
+class DecimalTransformer(ValueTransformer):
+    def transform(self, input_value: InputTypes) -> str:
```

Comment:
I like the general idea, but not the way how you put it into practice. These classes are purely statical, because they don't hold any information. In their current state, it doesn't make sense to have several objects (instantiations) from them. You can use the function directly with `DecimalTransformer.transform(val)`, without needing an actual `DecimalTransformer` object.

In other words, the only thing that the `DecimalTransformer` class does is encapsulating a method. But the encapsulation is effectively useless. It would make more sense if the method would be a free-standing function.

To demonstrate it, I have prepared this PR: https://github.com/dasch-swiss/dsp-tools/pull/1281. Feel free to modify it, merge it, or just let you inspire by it.